tsdoc/src/parser/NodeParser.ts

```diff
@@ -334,9 +334,13 @@ export class NodeParser {
     }
   }
 
+  /**
+   * Used by `_parseParamBlock()`, this parses a JSDoc expression remainder like `string}` or `="]"]` from
+   * an input like `@param {string} [x="]"] - the X value`.  It detects nested balanced pairs of delimiters
+   * and escaped string literals.
+   */
   private _tryParseJSDocTypeOrValueRest(tokenReader: TokenReader, openKind: TokenKind, closeKind: TokenKind): TokenSequence | undefined {
     const startMarker: number = tokenReader.createMarker();
```

Comment:
To ensure you backtrack to the correct position, I'd move `startMarker` to a parameter like I did in https://github.com/microsoft/tsdoc/pull/219

Reply:
Good catch.  My version did not report any error for this input:
```ts
/**
 * @param {x - The first input number
 */
```